dvc/tree/azure.py

```diff
@@ -36,6 +38,59 @@ def __init__(self, repo, config):
             container = self._az_config.get("storage", "container_name", None)
             self.path_info = self.PATH_CLS(f"azure://{container}")
 
+        self._get_string_token_credential(config)
+
+        if not (self._conn_str or self._credential):
+            # from azure.identity import DefaultAzureCredential
+            from azure.identity import (
+                AzureCliCredential,
+                ChainedTokenCredential,
+            )
+
+            account_name = config.get("account_name")
+            self._account_url = f"https://{account_name}.blob.core.windows.net"
+            if not account_name:
+                logging.warning(
+                    "You must enter an account name for this auth flow"
+                )
+
+            if config.get("azcli_credential"):
+                self._credential = ChainedTokenCredential(AzureCliCredential())
+            else:
+                self._credential = self._create_client_secret_credential(
+                    config
+                )
```

Comment:
Btw, this whole logic is coming from azcli, right? If you've used specific azcli sources, please consider putting a link here as a reference for the future. :slightly_smiling_face:

Reply:
Added a link to a page that would explain the library I've brought in.